File: app/src/main/java/com/semanienterprise/softcom/retrofit/service/ApiService.java
package com.semanienterprise.softcom.retrofit.service;

import com.semanienterprise.softcom.models.FormData;

import io.reactivex.Single;
import retrofit2.http.GET;
import retrofit2.http.Headers;
import retrofit2.http.POST;
import retrofit2.http.Path;

public interface ApiService {

    //TODO: Set the url to get the JSON Object
    //Api for getting JSON
    @Headers("Content-Type: application/json")
    @GET("Softcom/{fileName}")
    Single<FormData> getJson(
            @Path("fileName") String fileName
    );
}

File: app/src/main/java/com/semanienterprise/softcom/retrofit/service/ApiService.kt
package com.semanienterprise.softcom.retrofit.service

import com.jakewharton.retrofit2.adapter.rxjava2.RxJava2CallAdapterFactory
import com.semanienterprise.softcom.models.FormData
import com.semanienterprise.softcom.retrofit.network.ApiClient

import io.reactivex.Single
import retrofit2.Retrofit
import retrofit2.converter.gson.GsonConverterFactory
import retrofit2.http.GET
import retrofit2.http.Headers
import retrofit2.http.POST
import retrofit2.http.Path

interface ApiService {

    //Api for getting JSON
    @GET("Softcom/{fileName}")
    fun getJson(@Path("fileName") fileName: String): Single<FormData>

    companion object Client {
        fun create(): ApiService {
            val retrofit = Retrofit.Builder()
                    .baseUrl("https://generalkolo.github.io/")
                    .addCallAdapterFactory(RxJava2CallAdapterFactory.create())
                    .addConverterFactory(GsonConverterFactory.create())
                    .build()

            return retrofit.create(ApiService::class.java)
        }
    }
}

